File: M4/Proyecto01G1/app/src/main/java/com/example/proyecto01g1/MainActivity2.java
package com.example.proyecto01g1;

import android.os.Bundle;
import android.view.WindowInsets;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;
import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;


public class MainActivity2 extends AppCompatActivity {
    Button btnSumar;
    Button btnRestar;
    Button btnMultiplicar;
    Button btnDividir;
    Button btnLimpiar;
    EditText txtNumero1;
    EditText txtNumero2;
    TextView txtResultado;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        // asociar variable con los elementos disponibles en la vista
        btnSumar = findViewById(R.id.btnSumar);
        btnRestar = findViewById(R.id.btnRestar);
        btnMultiplicar = findViewById(R.id.btnMultiplicar);
        btnDividir = findViewById(R.id.btnDividir);
        btnLimpiar = findViewById(R.id.btnLimpiar);
        txtNumero1 = findViewById(R.id.txtNumero1);
        txtNumero2 = findViewById(R.id.txtNumero2);
        txtResultado = findViewById(R.id.txtResultado);


        btnSumar.setOnClickListener(view -> {
            int numero1 = Integer.parseInt(txtNumero1.getText().toString());
            int numero2 = Integer.parseInt(txtNumero2.getText().toString());
            int resultado = numero1 + numero2;
            txtResultado.setText("Resultado: " + resultado);
            txtNumero2.getWindowInsetsController().hide(WindowInsets.Type.ime());
            Toast.makeText(this, "Resultado: " + resultado, Toast.LENGTH_SHORT).show();
        });

        // Ejercicio 1: Realizar las otras operaciones aritmeticas.
        btnRestar.setOnClickListener(view -> {
            int numero1 = Integer.parseInt(txtNumero1.getText().toString());
            int numero2 = Integer.parseInt(txtNumero2.getText().toString());
            int resultado = numero1 - numero2;
            txtResultado.setText("Resultado: " + resultado);
            txtNumero2.getWindowInsetsController().hide(WindowInsets.Type.ime());// oculta el teclado
        });
        btnMultiplicar.setOnClickListener(view -> {
            int numero1 = Integer.parseInt(txtNumero1.getText().toString());
            int numero2 = Integer.parseInt(txtNumero2.getText().toString());
            int resultado = numero1 * numero2;
            txtResultado.setText("Resultado: " + resultado);
            txtNumero2.getWindowInsetsController().hide(WindowInsets.Type.ime());// oculta el teclado
        });
        btnDividir.setOnClickListener(view -> {
            int numero1 = Integer.parseInt(txtNumero1.getText().toString());
            int numero2 = Integer.parseInt(txtNumero2.getText().toString());
            txtNumero2.getWindowInsetsController().hide(WindowInsets.Type.ime());// oculta el teclado

            if (numero2 == 0) {
                Toast.makeText(this, "No se puede dividir por cero", Toast.LENGTH_SHORT).show();
                txtResultado.setText("");
                return;
            }
            // Tarea: buscar las diferentes sintaxis de la sentencia if
            // Tarea: Agregar al IDE, una IA (gemini u otra que se pueda agregar)
            int resultado = numero1 / numero2;
            txtResultado.setText("Resultado: " + resultado);
        });
        btnLimpiar.setOnClickListener(view -> {
            txtNumero1.setText(""); // limpia los text
            txtNumero2.setText("");
            txtResultado.setText("");
            txtNumero2.getWindowInsetsController().hide(WindowInsets.Type.ime()); // oculta el teclado
        });


    }
}

File: M5/ProyectoDonaciones/app/src/main/java/com/example/proyectodonaciones/view/MainActivity2.kt
package com.example.proyectodonaciones.view

import android.os.Bundle
import android.widget.TextView
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import com.example.proyectodonaciones.R
import com.example.proyectodonaciones.model.DonacionModelo
import com.example.proyectodonaciones.model.DonationModel

class MainActivity2 : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main2)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }

        var txtValor = findViewById<TextView>(R.id.txtValor)
        txtValor.text = DonacionModelo.totalAmount.toString()

        var txtValorRecibido = findViewById<TextView>(R.id.txtValorRecibido)
        txtValorRecibido.text = intent.getIntExtra("dato", 0).toString()


    }
}